W-15585387 Add troubleshooting Azure guide. (#384)
mulesoft/docs-rpa: rpa-builder/6.6/modules/ROOT/pages/troubleshooting-azure-oauth-setup.adoc

= Troubleshooting Microsoft Azure Setup

If you have issues configuring your Azure account with Mail Session (Outlook with OAuth), follow these troubleshooting steps to ensure your app is properly registered and configured:

. <<register-app>>
. <<create-secret>>
. <<verify-settings>>
. <<add-permissions>>

[[register-app]]
== Register an App in Azure

. Navigate to https://portal.azure.com/ and log in with your Azure account. 
. Open the *App Registrations* page and create a new registration. 
** Select the supported account types. 

> ** Provide a Redirect URI.

Reviewer: This is a regular point of confusion for customers, I think we should emphasize that the URL needs to be any valid URL and should provide an example like http://localhost:49453/

= Troubleshooting Microsoft Azure Setup

If you have issues configuring your Azure account with Mail Session (Outlook with OAuth), follow these troubleshooting steps to ensure your app is properly registered and configured:

. <<register-app>>
. <<create-secret>>
. <<verify-settings>>
. <<add-permissions>>

[[register-app]]
== Register an App in Azure

. Navigate to https://portal.azure.com/ and log in with your Azure account. 
. Open the *App Registrations* page and create a new registration. 
** Select the supported account types. 
** Provide a Redirect URI.
+
For the configuration to work, you must specify a valid redirect URL, for example: `+http://localhost:49453/+`.

For additional instructions about registering apps in Azure, see https://learn.microsoft.com/en-us/azure/active-directory-b2c/tutorial-register-applications#register-a-web-application[Azure: Register a web application].

[[create-secret]]
== Create a Client Secret for the App

After registering an app in Azure, the app's setting page opens.

. Open the *Certificates & Secrets* page.
. Create a new client secret.
+
[WARNING]

> After creating the secret, ensure you copy its *Value* as this is the only time you can see it. After you leave this page, the secret is not accessible anymore. 

Reviewer 2: ```suggestion
After creating the secret, ensure you copy its *Value* as this is the only time you can see it. After you leave this page, the secret is no longer accessible. 
```

[[verify-settings]]
== Verify the Settings

After registering the app and creating a client secret, open the app's *Overview* page and ensure you have: 

* One secret under *Client Credentials*
* One *Redirect URI*

If you created a Single-tenant app, locate the tenant ID for further use in the Mail Session (Outlook with OAuth) configuration. 

Depending on the account type you selected when registering your app (Single or Multi tenant), the *OAuth Host* you configure in Mail Session (Outlook with OAuth) changes:

* Multitenant: Specify `+https://login.microsoftonline.com/common/oauth2/v2.0/+` as the *OAuth Host* URL.
* Single tenant: Specify `+https://login.microsoftonline.com/{{YOUR_TENANT_ID}}/oauth2/v2.0/+` as the *OAuth Host* URL.

[[add-permissions]]
== Add Permissions to the App 

After registering your app and creating its client secret, configure permission settings:

. Open *API Permissions* in the side bar. 
. Click *Add a Permission* and select *Microsoft Graph*. 
. Select *Delegated Permissions* and add the following permissions from the *OpenId permissions* category: 
** `email`
** `offline_access`
** `openid`
** `profile`

> . Select the appropriate permissions and protocols depending on your Workflow needs: 

Reviewer 2: ```suggestion
. Select the appropriate permissions and protocols depending on your workflow needs: 
```

Reviewer 2: If "Workflow" is the name of a feature, dismiss this suggestion.

+
--
** Category `Mail`
** Category `IMAP`
** Category `EWS`
** Category `POP`
** Category `SMTP`
--
+

> If you are not sure which of these apply to your case, select all of them. 

Reviewer 2: ```suggestion
If you don't know the permissions and protocols that apply to your case, select all of them. 
```
